src/composer/polynomial.rs

```diff
@@ -0,0 +1,46 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use dusk_bls12_381::BlsScalar;
+
+use crate::constraint_system::Witness;
+
+/// Represents a polynomial in coefficient form with its associated wire data
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct Polynomial {
```

Comment:
We can't name it constraint because it is already taken. We can tweak this name in some further improvements

https://github.com/dusk-network/plonk/issues/706